core/app/c/[communitySlug]/pubs/[pubId]/components/jsonSchemaHelpers.tsx

```diff
@@ -60,21 +60,15 @@ export function recursivelyGetScalarFields(schema: JSONSchemaType<AnySchema>, va
 	return fields;
 }
 
-export function renderField(fieldValue: PubValueWithFieldAndSchema) {
-	const JSONSchema = fieldValue.field.schema
-		? (fieldValue.field.schema.schema as JSONSchemaType<AnySchema>)
-		: null;
-	const fieldTitle = (JSONSchema && JSONSchema.title) || fieldValue.field.name;
-	const renderedField = JSONSchema
-		? recursivelyGetScalarFields(JSONSchema, fieldValue.value)
-		: fieldValue.value && fieldValue.value.toString();
```

Comment:
I simplified this function to make the arguments it takes easier to work with. this whole thing will likely have to be reworked when we figure out how individual pub pages should be rendered. in any case, this part that looks at the `JSONSchema` wasn't being used since we use `schemaName` over `schema` now, so everything was always rendered as `toString()`

Reply:
i agree that we need to rework it, but im afraid this might break some of unjournals stuff maybe? I'm not sure im up to date on where we stand wrt to their migration

Reply:
I didn't want to add a query for the deprecated `schema` field to `getPubsWithRelatedValuesAndChildren`, so @kalilsn checked the data for me ([convo here](https://knowledgefutures.slack.com/archives/CMAQM0BNV/p1732732811633689)) and it seems like we only have to worry about file upload previews. so I [updated this code](https://github.com/pubpub/platform/pull/810/commits/a4ff35dcc357ae60811247b3445d977ffcc6a7ed) to check if the value looks like a file upload, and if so then to render the preview. everything else continues to render as strings